lib/Doctrine/DBAL/Platforms/AbstractPlatform.php

```diff
@@ -3525,4 +3525,17 @@ public function getStringLiteralQuoteCharacter()
     {
         return "'";
     }
+
+    /**
+     * Determines if a string literal is valid as an identifier in this
+     * platform.
+     *
+     * @param string $name The string literal to test
+     *
+     * @return bool
+     */
+    public function isValidIdentifier($name)
```

Comment:
Yes, an internal approach is fine here - the only requirement is for the DBAL to produce a valid schema. This functionality can just as easily be rolled back into SQLServerSchemaManager.

If that seems like a good course of action I can take care of that in short order.